pkg/scheduler/apis/config/validation/validation_pluginargs_test.go

```diff
@@ -850,6 +850,53 @@ func TestValidateNodeResourcesMostAllocatedArgs(t *testing.T) {
 	}
 }
 
+func TestValidateNodeResourcesBalancedAllocationArgs(t *testing.T) {
+	cases := map[string]struct {
+		args     *config.NodeResourcesBalancedAllocationArgs
+		wantErrs field.ErrorList
+	}{
+		"valid config": {
+			args: &config.NodeResourcesBalancedAllocationArgs{
+				Resources: []config.ResourceSpec{
+					{
+						Name:   "cpu",
+						Weight: 1,
+					},
+					{
+						Name:   "memory",
+						Weight: 1,
+					},
+				},
+			},
+		},
+		"weight is not 1": {
+			args: &config.NodeResourcesBalancedAllocationArgs{
+				Resources: []config.ResourceSpec{
+					{
+						Name:   "cpu",
+						Weight: 10,
+					},
+				},
+			},
+			wantErrs: field.ErrorList{
+				&field.Error{
+					Type:  field.ErrorTypeInvalid,
+					Field: "resources[0].weight",
+				},
+			},
+		},
+	}
```

Comment:
Can you refactor the tests to cover:

- multiple resources
  - only one has illegal weight
  - more than one have illegal weight (expect to see more than one error)

Reply:
sure, will do

Reply:
new type won't have a weight.